spec/support/shared_examples/step_controller.rb

```diff
@@ -9,6 +9,18 @@
 
       expect(step).to be_an_instance_of step_class
     end
+
+    context "no application is currently under way" do
+      it "redirects to the homepage" do
+        if subject.class != IntroduceYourselfController
+          session[:snap_application_id] = nil
+
+          get :edit
+
+          expect(response).to redirect_to(root_path)
+        end
```

Comment:
If I remove this:

```
def ensure_application_present
  _application_not_created = true
end
```

... then you'll never be able to get into the first step. It'll always redirect you back.